openeo/rest/connection.py

```diff
@@ -842,13 +842,13 @@ def load_collection(
         """
         if self._api_version.at_least("1.0.0"):
             return DataCube.load_collection(
-                collection_id=collection_id, connection=self,
+                collection_id=id, connection=self,
```

Comment:
I think the `load_collection` methods of DataCube and ImageCollectionClient should also be addressed

but here it's fine to do it in a backwards incompatible way because these are practically "internal" methods nobody should be using